stackslib/src/net/api/postblock_proposal.rs

```diff
@@ -757,6 +670,190 @@ impl NakamotoBlockProposal {
+                    // the block contains a transaction that is not in the replay set,
+                    // and we should reject the block.
+                    warn!("Rejected block proposal. Block contains transactions beyond the replay set.";
+                        "txid" => %tx.txid(),
+                        "tx_index" => i,
+                    );
+                    return Err(BlockValidateRejectReason {
+                        reason_code: ValidateRejectCode::InvalidTransactionReplay,
+                        reason: "Block contains transactions beyond the replay set".into(),
+                    });
+                };
+                if replay_tx.txid() == tx.txid() {
+                    break;
+                }
+
+                // The included tx doesn't match the next tx in the
+                // replay set. Check to see if the tx is skipped because
+                // it was unmineable.
+                let tx_result = replay_builder.try_mine_tx_with_len(
+                    &mut replay_tenure_tx,
+                    &replay_tx,
+                    replay_tx.tx_len(),
+                    &BlockLimitFunction::NO_LIMIT_HIT,
+                    ASTRules::PrecheckSize,
+                    None,
+                );
+                match tx_result {
+                    TransactionResult::Skipped(TransactionSkipped { error, .. })
+                    | TransactionResult::ProcessingError(TransactionError { error, .. })
+                    | TransactionResult::Problematic(TransactionProblematic { error, .. }) => {
+                        // The tx wasn't able to be mined. Check the underlying error, to
+                        // see if we should reject the block or allow the tx to be
+                        // dropped from the replay set.
+
+                        match error {
+                            ChainError::CostOverflowError(..)
+                            | ChainError::BlockTooBigError
+                            | ChainError::ClarityError(ClarityError::CostError(..)) => {
+                                // block limit reached; add tx back to replay set.
+                                // BUT we know that the block should have ended at this point, so
+                                // return an error.
+                                let txid = replay_tx.txid();
+                                replay_txs.push_front(replay_tx);
+
+                                warn!("Rejecting block proposal. Next replay tx exceeds cost limits, so should have been in the next block.";
+                                    "error" => %error,
+                                    "txid" => %txid,
+                                );
+
+                                return Err(BlockValidateRejectReason {
+                                    reason_code: ValidateRejectCode::InvalidTransactionReplay,
+                                    reason: "Transaction is not in the replay set".into(),
```

Comment:
Looks like this was there before this refactoring, but this reason should be updated.